python/ray/tests/kuberay/test_autoscaling_e2e.py

```diff
@@ -0,0 +1,232 @@
+import logging
+import os
+import pathlib
+import tempfile
+import unittest
+import subprocess
+
+from typing import Any, Dict
+
+import yaml
+
+from ray.tests.kuberay.utils import (
+    get_pod,
+    kubectl_exec,
+    wait_for_pods,
+    wait_for_pod_to_start,
+    wait_for_ray_health,
+    wait_for_crd,
+)
+
+logger = logging.getLogger(__name__)
+logging.basicConfig(
+    level=logging.INFO,
+    format="[%(levelname)s %(asctime)s] " "%(filename)s: %(lineno)d  " "%(message)s",
+)
+
+# This image will be used for both the Ray nodes and the autoscaler.
+# The CI should pass an image built from the test branch.
+RAY_IMAGE = os.environ.get("RAY_IMAGE", "rayproject/ray:413fe0")
+logger.info(f"Using image {RAY_IMAGE} for autoscaler and Ray nodes.")
+# The default "rayproject/ray:413fe0" is the currently pinned autoscaler image
+# (to be replaced with rayproject/ray:1.12.0 upon 1.12.0 release).
+
+# Parent directory of Ray repository
+RAY_PARENT = str(pathlib.Path(__file__).resolve().parents[5])
+# Path to example config rel RAY_PARENT
+EXAMPLE_CLUSTER_PATH = "ray/python/ray/autoscaler/kuberay/ray-cluster.complete.yaml"
+
+
+class KubeRayAutoscalingTest(unittest.TestCase):
+    """e2e verification of autoscaling following the steps in the Ray documentation.
+    kubectl is used throughout, as that reflects the instructions in the docs.
+    """
+
+    def setUp(self):
+        """Set up KubeRay operator and Ray autoscaler RBAC."""
+
+        # Switch to parent of Ray repo, because that's what the doc examples do.
+        logger.info("Switching to parent of Ray directory.")
+        os.chdir(RAY_PARENT)
+
+        logger.info("Cloning KubeRay and setting up KubeRay configuration.")
+        subprocess.check_call(
+            [
+                "bash",
+                "-c",
+                "ls kuberay || ./ray/python/ray/autoscaler/kuberay/init-config.sh",
+            ]
+        )
+        logger.info("Creating KubeRay operator.")
+        subprocess.check_call(
+            [
```

Comment:
regardign these common kubectl and kind operations, we may also reuse them in our kuberay tests. So putting them together in a bunch of common files would help. Anyway, this can be something that we do later.

Reply:
From the meeting, perhaps we should think more carefully about what the utils we want to pull out will look like to avoid versioning issues between the ci's (and therefor avoid making this change in this PR)

Reply:
As we expand this Ray repo's tests, we might consider pulling more stuff into the utils file that's already there.